File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def choose(idx, num_sum):
        if idx == len(numbers):
            if num_sum == target:
                return 1
            return 0
        return choose(idx + 1, num_sum + numbers[idx]) + choose(idx + 1, num_sum - numbers[idx])

    answer = choose(0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer, target_g;
    static int[] numbers_g;
    
    public int solution(int[] numbers, int target) {
        answer = 0;
        target_g = target;
        numbers_g = numbers;
        
        choose(0, 0);
        
        return answer;
    }
    
    public static void choose(int count, int total){
        if(count == numbers_g.length){
            if(total == target_g)
                answer++;
            return;
        }
        
        choose(count + 1, total+numbers_g[count]);
        choose(count + 1, total-numbers_g[count]);
    }
}